### ISSUE TITLE
[Payment history] Investigate continued latency issues

### ISSUE THREAD

**Issue by Samara-Strauss** [department-of-veterans-affairs/va.gov-team#41511]
### Background

We keep seeing an issue around payment history latency. On Sunday night (5/8/22), there was about a 4 hour period where there was a huge spike in errors for both payment history and ch 33 direct deposit benefits that dropped off as suddenly as it started. It's still unclear to us what caused this issue, but it is possible this was due to BGS downtime.

Then, around 10:00 AM ET yesterday morning (5/9/22), we started getting latency alerts for payment history, Ch 33 direct deposit benefits, and debt (note: we know your team only manages payment history and not the other two endpoints). This lasted for a few hours, and then eventually resolved on its own around 6:30 pm ET on 5/9/22.

Unfortunately, this issue occurred again yesterday (5/16/22) in a similar pattern. On Sunday night (5/15/22), there was about a 4 hour period (11pm - 3am) where there was a significant spike in errors for both payment history. Then, around 1:45 PM ET on 5/16/22, we started getting latency alerts for payment history. The latency has since cleared as of 5:45 AM ET on 5/17/22.

We're not sure what's going on here, but it looks like a pattern is starting to develop which may help shed some light on what is causing the issue. We wanted to flag this for the benefits team since you all manage payment history, but we have also created a devops issue in #41123 in case this is something that is more in their purview.

### Tasks

- [ ] Track latency to determine if there is a pattern with payment history
- [ ] Work with devops and authenticated experience teams as needed to determine why these latency alerts keep happening

**Comment by saderagsdale** [comment]
@cohnjesse Can you assign this to a developer to review?

**Comment by cohnjesse** [comment]
@saderagsdale yep. Though it mentioned this is not just an issue with our code, it affects multiple apps, so this may be a Ops issue

**Comment by saderagsdale** [comment]
@jacobworrell adding that squad 1 label to this.

**Comment by michelpmcdonald** [comment]
Leaving this open right now.....I'm not aware of any request from  BGS for more information from me to help them troubleshoot their endpoint.

**Comment by jacobworrell** [comment]
Here's our recommended course of action: 

- Look into the Grafana logs (or other performance tools) and determine what data we have that corresponds to the time and dates of the latency issues. Provide additional context/information relevant to Samara's investigation. 

@cohnjesse is this an action you'd be willing to take on behalf of Squad 1?

**Comment by cohnjesse** [comment]
@saderagsdale @Samara-Strauss so since this is an issue with BGS we need to get them the information they need in order to isolate the issue and fix it. @Samara-Strauss what information does BGS need from us? I am happy to get that information over to them and let them handle this.

**Comment by Samara-Strauss** [comment]
Bringing @tpharrison into this, since he's been helping field information to BGS as well.

The last time this happened (earlier this week), BGS finally started to pay a little more attention and asked us to let them know the next time this happens while it's happening (which we usually do). We send them over which endpoints and the start time of the latency alert. We eventually send over end times as well, as the issue usually clears itself.

@tpharrison am I missing anything?

**Comment by tpharrison** [comment]
@Samara-Strauss - you covered it.  We've provided all the information BGS has requested.  Over the past couple of weeks, the latency issue seems to occur less frequently and when it does, the timeframe is shorter.   The last two times there was significant latency, it lasted for about 1 hour. 

As Samara stated, BGS has requested that we reach out to them during the next occurrence.  Our latency alerts are based on average response times so there is normally about a 1 hour delay between when it starts and when we're alerted.  I've adjusted this so that we'll be notified sooner.
